src/main/java/taxi/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,57 @@
+package taxi.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.ManufacturerDao;
+import taxi.db.Storage;
+import taxi.exception.NoSuchManufacturerException;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        return Storage.manufacturers
+                .stream()
+                .filter(man -> man.getId().equals(id))
+                .findAny();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Optional<Manufacturer> targetManufacturer = get(manufacturer.getId());
+        for (Manufacturer storageManufacturer : Storage.manufacturers) {
+            if (targetManufacturer.isPresent()) {
+                if (storageManufacturer.getId().equals(targetManufacturer.get().getId())) {
```

Comment:
use for i loop or stream and method list.set() when you found the needed index instead. This way if any field has been changed or you added a field you won't need to rewrite this code

Reply:
refactored to use an I loop with `list.set()` method. Looks much better now, thank you.